omim2obo/main.py

```diff
@@ -243,6 +241,12 @@ def run(use_cache: bool = False):
             chr_id = '9606chr' + cyto_location
             # What's RO['0002525'] - joeflack4 2021/11/11
             graph.add((OMIM[mim_number], RO['0002525'], CHR[chr_id]))
+        if mim_data['phenotype_mapping_info_label']:
+            # todo: BioLink has a formal way to do this, which we've opted not to do for now. But perhaps in the future
+            #  we can do it. It would involve first the (OMIM[mim_number], BIOLINK['GeneToDiseaseAssociation'],
+            #  BIOLINK['has_evidence']), followed by some way to annotate w/ mim_data['phenotype_mapping_info_label']
+            #  https://biolink.github.io/biolink-model/docs/GeneToDiseaseAssociation.html
+            graph.add((OMIM[mim_number], BIOLINK['has_evidence'], Literal(mim_data['phenotype_mapping_info_label'])))
```

Comment:
Related if you want to go down the SPARQL route: https://github.com/monarch-initiative/mondo-ingest/blob/0b255f5ded749a424fcc072444e672833875ed37/src/sparql/fix_illegal_punning_omim.ru

Reply:
Also related to #60

Reply:
These are great suggesitons. I would do this the Python rdflib route rather than SPARQL after the fact.

Before I make these changes, let's discuss further, in light of a few things:
1. How to add multiple gene::phenotype associations
2. What to do when there is no phenotype MIM#

Reply:
Done! Example:
```owl
OMIM:100678 a owl:Class ;
    rdfs:label "ACAT2" ;
    ...
    rdfs:subClassOf [ a owl:Restriction ;
            owl:onProperty RO:0003303 ;
            owl:someValuesFrom OMIM:614055 ],
        SO:0000704 ;
    biolink:category biolink:Gene .
```

```owl
[] a owl:Axiom ;
    rdfs:comment "Evidence: (1) The disorder is placed on the map based on its association with a gene, but the underlying defect is not known." ;
    owl:annotatedProperty rdfs:subClassOf ;
    owl:annotatedSource OMIM:100678 ;
    owl:annotatedTarget [ a owl:Restriction ;
            owl:onProperty RO:0003303 ;
            owl:someValuesFrom OMIM:614055 ] .
```

Reply:
@matentzn Now, multiple gene::disease associations are being added in this way. I need you to check this:

Class snippet. Note that there are 3 identical subclass bnodes with refs to `OMIM:266300`:
```ttl
OMIM:155555 a owl:Class ;
    rdfs:subClassOf [ a owl:Restriction ;
            owl:onProperty RO:0003302 ;
            owl:someValuesFrom OMIM:266300 ],
        [ a owl:Restriction ;
            owl:onProperty RO:0003302 ;
            owl:someValuesFrom OMIM:203200 ],
        [ a owl:Restriction ;
            owl:onProperty RO:0003302 ;
            owl:someValuesFrom OMIM:266300 ],
        [ a owl:Restriction ;
            owl:onProperty RO:0003302 ;
            owl:someValuesFrom OMIM:613099 ],
        [ a owl:Restriction ;
            owl:onProperty RO:0003302 ;
            owl:someValuesFrom OMIM:613098 ],
        [ a owl:Restriction ;
            owl:onProperty RO:0003302 ;
            owl:someValuesFrom OMIM:266300 ],
```

For each of the 3 `OMIM:266300` refs above, there are 3 of these identical axioms now in the output:
```ttl
[] a owl:Axiom ;
    rdfs:comment "Evidence: (3) The molecular basis for the disorder is known; a mutation has been found in the gene." ;
    owl:annotatedProperty rdfs:subClassOf ;
    owl:annotatedSource OMIM:155555 ;
    owl:annotatedTarget [ a owl:Restriction ;
            owl:onProperty RO:0003302 ;
            owl:someValuesFrom OMIM:266300 ] ;
    biolink:has_evidence "Evidence: (3) The molecular basis for the disorder is known; a mutation has been found in the gene." .
```

## Questions
1. Do we want to add the `Phenotype` label so that these do not appear as duplicates? Note that they are not duplicates in `morbidmap.txt`. They just look like duplicates here because we leave out that `Phenotype` label, which differs between the 3. The labels are the only difference between these 3 instances.

Here are the original rows which include the labels:
```tsv
Phenotype	Gene Symbols	MIM Number	Cyto Location
[Skin/hair/eye pigmentation 2, blond hair/fair skin], 266300 (3)	MC1R, SHEP2, CMM5	155555	16q24.3
[Skin/hair/eye pigmentation 2, red hair/fair skin], 266300 (3)	MC1R, SHEP2, CMM5	155555	16q24.3
{UV-induced skin damage}, 266300 (3)	MC1R, SHEP2, CMM5	155555	16q24.3
```

If we leave out the label, I can dedupe these structures. It is lossy, but I don't know if that matters at this point.

Reply:
- The fact that the exact same "axiom" can appear twice is just a consequence of rdflib - once this is cycled through ROBOT, which uses a set-based system for statements, these will all collapse into 1. No need to write code for deduplication!
- Can you remind me (sorry, too many things for my tiny brain) what the phrase `Do we want to add the Phenotype label?` means concretely? Can you give an example?

Reply:
Ah, I see. So the OWL `.ttl` that's being output by `rdflib` is invalid. I could certainly tweak my code to avoid this duplication, but given that it's going through `robot`, let's handle it that way for now as you suggested, thanks!

---

Regarding "phenotype label", I can help elaborate on what I mean. Consider the second of the 3 rows above:
> `[Skin/hair/eye pigmentation 2, red hair/fair skin], 266300 (3)`

This `Phenotype` field consists of 3 parts:
i. Label: `[Skin/hair/eye pigmentation 2, red hair/fair skin]`
ii. MIM#: `, 266300`
iii. Mapping evidence key: ` (3)`

At least, 'label' is what I call it. I suppose you could also call it a 'title' or 'name'.

Reply:
Ah, I see. So the OWL `.ttl` that's being output by `rdflib` is invalid. I could certainly tweak my code to avoid this duplication, but given that it's going through `robot`, let's handle it that way for now as you suggested, thanks!

---

Regarding "phenotype label", I can help elaborate on what I mean. Consider the second of the 3 rows above:
> `[Skin/hair/eye pigmentation 2, red hair/fair skin], 266300 (3)`

This `Phenotype` field consists of 3 parts:
i. Label: `[Skin/hair/eye pigmentation 2, red hair/fair skin]`
ii. MIM#: `, 266300`
iii. Mapping evidence key: ` (3)`

At least, 'label' is what I call it. I suppose you could also call it a 'title' or 'name'.

Reply:
> Do we want to add the Phenotype label so that these do not appear as duplicates?

Ok now that I understand what you mean, i am confused: why would there ever be a different label for the same MIM number?

Reply:
@matentzn You tell me Nico! @sabrinatoro @nicolevasilevsky Maybe you guys know: Why do we think that `morbidmap.txt` provides different labels for the same MIM# within the `Phenotype` column, as below?

```tsv
Phenotype	Gene Symbols	MIM Number	Cyto Location
[Skin/hair/eye pigmentation 2, blond hair/fair skin], 266300 (3)	MC1R, SHEP2, CMM5	155555	16q24.3
[Skin/hair/eye pigmentation 2, red hair/fair skin], 266300 (3)	MC1R, SHEP2, CMM5	155555	16q24.3
```

If we don't know the answer to this, I can send an email query to my OMIM contacts.

Reply:
we have been discussing this in some issues. i dont remember at the moment what our planned action is though

Reply:
the open issues for this have the apporpriate discussion on follow up steps